diff --git a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
index 5a5a6ac2b0e..08f44d253a7 100644
--- a/src/amd/common/ac_gpu_info.c
+++ b/src/amd/common/ac_gpu_info.c
@@ -1547,7 +1547,7 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
 
    ac_fill_feature_info(info, &device_info);
 
-   info->kernel_has_modifiers = has_modifiers(fd) || (info->is_virtio && fd < 0);
+   info->kernel_has_modifiers = has_modifiers(fd) || info->is_virtio;
 
    ac_fill_bug_info(info);
 
